src/Renderer/UI/SimulationView.fs

```diff
@@ -378,23 +384,148 @@ let private viewStatefulComponents step comps numBase model dispatch =
         | _ -> []
     div [] (List.collect makeStateLine comps )
 
-let viewSimulationError (simError : SimulationError) =
-    let error = 
-        match simError.InDependency with
-        | None ->
+let viewSimulationError (comps: Component list, conns: Connection list) (simError : SimulationError) (model: Model) endDispatch dispatch =
+    let sheetDispatch sMsg = dispatch <| Sheet sMsg
+    let busWireDispatch bMsg = sheetDispatch <| SheetT.Msg.Wire bMsg
+    let symbolDispatch symMsg = busWireDispatch <| BusWireT.Msg.Symbol symMsg
+
+    let tryGetComponentById compId =
+        comps
+        |> List.tryFind (fun comp -> comp.Id = compId)
+    
+    let tryGetConnectionById connId =
+        conns
+        |> List.tryFind (fun conn -> conn.Id = connId)
+    
+    let reacListOfCompsAffected =
+        simError.ComponentsAffected
```

Comment:
this is juts filter followed by map where you filter on component id existing or something.
Lits.collect (Option.toList try... ) has the smell of something simplifiable (and is obscure)